src/jit/emit.h

```diff
@@ -884,8 +884,8 @@ class emitter
         }
         void idCodeSize(unsigned sz)
         {
+            assert(sz <= 15); // Intel decoder limit.
             _idCodeSize = sz;
-            assert(sz == _idCodeSize);
```

Comment:
It doesn't seem necessary (or maybe even desirable?) to remove this assert. It's a common pattern to have this assert when we're storing to a bitfield. Yes, your new assert is currently sufficient, but what if someone incorrectly changes the bitfield size to 3 bits?

Reply:
Do you think we should save both (new and old) or only the old one?

Reply:
I would leave both.